include/boost/gil/extension/numeric/convolve.hpp

```diff
@@ -274,7 +274,7 @@ void convolve_cols(
 /// \tparam DstView Models MutableImageViewConcept
 template <typename PixelAccum, typename SrcView, typename Kernel, typename DstView>
 BOOST_FORCEINLINE
-void convolve(
+void convolve_1d(
```

Comment:
@miralshah365 This is an interesting issue: it breaks compatibility, so older clients will no longer be able to compile with future Boost. For compatibility reasons, it's safer to keep `convolve` which means 1D and `convolve_2d` which means 2D.

@stefanseefeld Would you have any comment on that, e.g. in case I'm overcautious.

Reply:
@miralshah365 I've just reminded myself the `convolve` function has been added very recently in #347 and not released yet. This change should not break any existing clients, other than those which use GIL from `develop` which is cutting edge, so breakage is possible.
It's all good.

p.s. It's good to be sensitive to this kind of changes though :-)

Reply:
I never had such thought in my mind of breaking the compatibility for an existing client. I would now keep such thing while changing any existing code.